addons/project_sale_expense/models/hr_expense.py

```diff
@@ -0,0 +1,14 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+
+
+class Expense(models.Model):
+    _inherit = "hr.expense"
+
+    def _compute_analytic_distribution(self):
+        super()._compute_analytic_distribution()
+        for expense in self.filtered('sale_order_id'):
```

Comment:
I would prefer doing that:

```suggestion
        for expense in self:
            if not expense.sale_order_id:
                continue
```
to avoid doing 2 loops (one for the filtered) and another for the for loop. 🙂